stylix/target.nix

```
@@ -34,28 +34,83 @@
   config.lib.stylix =
     let
       cfg = config.stylix;
+      self = config.lib.stylix;
+
+      # Will wrap with (parentheses) if the expr contains operators with lower precedence than `&&`
+      wrapExprWith =
+        {
+          autoWrapExpr ? true,
+          trimExpr ? true,
+        }:
+        expr:
+        let
+          trim = if trimExpr then lib.strings.trim else lib.id;
+          trimmed = trim expr;
+          isWrapped = builtins.match ''[(].*[)]'' trimmed != null;
+          needsWrapping = builtins.any (op: lib.strings.hasInfix op trimmed) [
+            # These operators have lower precedence than `&&`
+            # See https://nix.dev/manual/nix/2.28/language/operators
+            "||"
+            "->"
+            "|>"
+            "<|"
+          ];
+        in
+        if autoWrapExpr && !isWrapped && needsWrapping then "(${trimmed})" else trimmed;
     in
     {
       mkEnableTarget =
-        humanName: autoEnable:
-        lib.mkEnableOption "theming for ${humanName}"
-        // {
-          default = cfg.autoEnable && autoEnable;
-          example = !autoEnable;
-        }
-        // lib.optionalAttrs autoEnable {
-          defaultText = lib.literalMD "same as `stylix.autoEnable`";
+        name: autoEnable:
+        config.lib.stylix.mkEnableTargetWith { inherit name autoEnable; };
+
+      mkEnableTargetWith =
+        {
+          name,
+          autoEnable ? true,
+          autoEnableExpr ? null,
+          autoWrapExpr ? true,
+          example ? if args ? autoEnableExpr then true else !autoEnable,
+        }@args:
+        let
+          wrapExpr = wrapExprWith {
+            inherit autoWrapExpr;
+          };
+        in
+        self.mkEnableIf {
+          description = "Whether to enable theming for ${name}";
+          condition = cfg.autoEnable && autoEnable;
+          conditionText =
+            if args ? autoEnableExpr then
+              lib.literalExpression "stylix.autoEnable && ${wrapExpr autoEnableExpr}"
+            else if autoEnable then
+              lib.literalMD "same as `stylix.autoEnable`"
```

Should we for consistency with

https://github.com/nix-community/stylix/blob/27b8efe7a0c68b8f7e5b2c00a77b28eded1cc97f/stylix/target.nix#L83-L84

get rid of "same as":

```suggestion
              lib.literalExpression "stylix.autoEnable"
```